messenger/src/pulsar_messenger.rs

```diff
@@ -201,10 +187,8 @@ impl Messenger for PulsarMessenger {
         let mut consumer = if let Some(consumer) = self.consumers.get(stream_key) {
             consumer.lock().await
```

Comment:
It does not allow to use `std::sync::Mutex`:

the trait `std::marker::Send` is not implemented for `std::sync::MutexGuard<'_, pulsar::Consumer<Vec<u8>, TokioExecutor>>`

Reply:
My mistake, you *are* holding onto the mutex across `await` statements, which I missed at first.  This makes it require `Send` because any time there is an `await`, the runtime can change threads, and `Send` trait indicates it can be sent to another thread.  So it is fine as-is.